lib/internal/test_runner/mock/mock.js

```diff
@@ -62,8 +64,63 @@ class MockFunctionContext {
     validateFunction(implementation, 'implementation');
     const nextCall = this.#calls.length;
     const call = onCall ?? nextCall;
-    validateInteger(call, 'onCall', nextCall);
-    this.#mocks.set(call, implementation);
+    // eslint-disable-next-line eqeqeq
+    if (onCall != undefined) {
+      validateInteger(call, 'onCall', nextCall);
+    }
+
+    // eslint-disable-next-line eqeqeq
+    if (onCall == undefined) {
+      this.#addMockToFirstEmptyImpl(implementation);
+      return this;
+    }
+
+    const lastCall = this.#queue[this.#queue.length - 1]?.index;
+    if (lastCall >= onCall) {
+      this.#replaceCallImplementation(onCall, implementation);
+    } else {
+      this.#addImplementationsPadding(lastCall, onCall);
+      this.#queue.push({ __proto__: null, index: onCall, impl: implementation });
+    }
+
+    return this;
+  }
+
+  #addMockToFirstEmptyImpl(implementation) {
+    if (this.#emptyImpl.size) {
+      const emptyImpl = this.#emptyImpl.values().next().value;
```

Comment:
```suggestion
      const [emptyImpl] = this.#emptyImpl;
```

Reply:
I can't do that is the eslint change to {0: emptyImpl} and that failed as it's not an array

Reply:
This is OK to use on `SafeSet`, so I think this is a false-positive of the lint rule... 🙁 

```shell
$ ./node --expose-internals -r internal/test/binding

> SafeSet = primordials.SafeSet
> Set.prototype[Symbol.iterator] = () => ({ next() { return { done: true }; } });
[Function (anonymous)]
> set = new SafeSet([1,2,3])
SafeSet(3) [Set] { 1, 2, 3 }
> [f] = set
SafeSet(3) [Set] { 1, 2, 3 }
> f
1
> set = new Set([1,2,3])
Set(3) {  }
> [f] = set
Set(3) {  }
> f
undefined
```

Anyway, let's leave it as is